operator/pkg/central/extensions/reconcile_tls_test.go

```diff
@@ -237,3 +374,379 @@ func TestRenewInitBundle(t *testing.T) {
+				mtls.CACertFileName:      ca1.CertPEM(),
+				mtls.CAKeyFileName:       ca1.KeyPEM(),
+				mtls.ServiceCertFileName: centralCertFromCA1.CertPEM,
+			},
+			assert: func(t *testing.T, err error) {
+				assert.ErrorContains(t, err, "no service private key in file map")
+			},
+		},
+		"should fail when the service cert does not match the service key": {
+			fileMap: types.SecretDataMap{
+				mtls.CACertFileName:      ca1.CertPEM(),
+				mtls.CAKeyFileName:       ca1.KeyPEM(),
+				mtls.ServiceCertFileName: centralCertFromCA1.CertPEM,
+				mtls.ServiceKeyFileName:  centralCertFromCA2.KeyPEM,
+			},
+			assert: func(t *testing.T, err error) {
+				assert.ErrorContains(t, err, "mismatched certificate and private key")
+			},
+		},
+		"should fail when the service cert is invalid": {
+			fileMap: types.SecretDataMap{
+				mtls.CACertFileName:      ca1.CertPEM(),
+				mtls.CAKeyFileName:       ca1.KeyPEM(),
+				mtls.ServiceCertFileName: []byte("invalid"),
+				mtls.ServiceKeyFileName:  centralCertFromCA1.KeyPEM,
+			},
+			assert: func(t *testing.T, err error) {
+				assert.ErrorContains(t, err, "unparseable certificate in file map")
+			},
+		},
+		"should fail when the service key is invalid": {
+			fileMap: types.SecretDataMap{
+				mtls.CACertFileName:      ca1.CertPEM(),
+				mtls.CAKeyFileName:       ca1.KeyPEM(),
+				mtls.ServiceCertFileName: centralCertFromCA1.CertPEM,
+				mtls.ServiceKeyFileName:  []byte("invalid"),
+			},
+			assert: func(t *testing.T, err error) {
+				assert.ErrorContains(t, err, "invalid private key")
+			},
+		},
+		"should fail when the service cert is not signed by the ca cert": {
+			fileMap: types.SecretDataMap{
+				mtls.CACertFileName:      ca1.CertPEM(),
+				mtls.CAKeyFileName:       ca1.KeyPEM(),
+				mtls.ServiceCertFileName: centralCertFromCA2.CertPEM,
+				mtls.ServiceKeyFileName:  centralCertFromCA2.KeyPEM,
+			},
+			assert: func(t *testing.T, err error) {
+				assert.ErrorContains(t, err, "certificate signed by unknown authority")
+			},
+		},
+		"should fail when the service cert subject is not the expected service name": {
+			fileMap: types.SecretDataMap{
+				mtls.CACertFileName:      ca1.CertPEM(),
+				mtls.CAKeyFileName:       ca1.KeyPEM(),
+				mtls.ServiceCertFileName: unexpectedSubjectCertFromCA1.CertPEM,
+				mtls.ServiceKeyFileName:  unexpectedSubjectCertFromCA1.KeyPEM,
+			},
+			assert: func(t *testing.T, err error) {
+				assert.ErrorContains(t, err, "unexpected certificate service type")
+			},
+		},
+		"should succeed when the ca cert and key are valid": {
+			fileMap: types.SecretDataMap{
+				mtls.CACertFileName:      ca1.CertPEM(),
+				mtls.CAKeyFileName:       ca1.KeyPEM(),
+				mtls.ServiceCertFileName: centralCertFromCA1.CertPEM,
+				mtls.ServiceKeyFileName:  centralCertFromCA1.KeyPEM,
+			},
+			assert: func(t *testing.T, err error) {
+				assert.NoError(t, err)
+			},
+		},
+		// TODO: Discuss tests around ca/service cert expiration, as well as "NotBefore".
```

Comment:
Good finding, but I fear this is something that could turn into a rathole..
How about filing a ticket about further improvements and putting its ID in `TODO()`?